components/brave_shields/browser/ad_block_subscription_service_manager.cc

```diff
@@ -0,0 +1,300 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_shields/browser/ad_block_subscription_service_manager.h"
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "base/strings/string_util.h"
+#include "base/task/post_task.h"
+#include "base/values.h"
+#include "brave/browser/brave_browser_process_impl.h"
+#include "brave/common/pref_names.h"
+#include "brave/components/brave_shields/browser/ad_block_subscription_service.h"
+#include "brave/components/brave_shields/browser/ad_block_service.h"
+#include "brave/components/brave_shields/browser/ad_block_service_helper.h"
+#include "brave/vendor/adblock_rust_ffi/src/wrapper.h"
+#include "components/prefs/pref_service.h"
+#include "components/prefs/scoped_user_pref_update.h"
+#include "content/public/browser/browser_task_traits.h"
+#include "content/public/browser/browser_thread.h"
+
+namespace brave_shields {
+
+AdBlockSubscriptionServiceManager::AdBlockSubscriptionServiceManager(
+    brave_component_updater::BraveComponent::Delegate* delegate)
+    : delegate_(delegate),
+      initialized_(false) {
+}
+
+AdBlockSubscriptionServiceManager::~AdBlockSubscriptionServiceManager() {
+}
+
+bool AdBlockSubscriptionServiceManager::Init() {
+  DCHECK(!initialized_);
+  base::PostTask(
+      FROM_HERE, {content::BrowserThread::UI},
+      base::BindOnce(&AdBlockSubscriptionServiceManager::StartSubscriptionServices,
```

Comment:
Why do we need that at all? `StartSubscriptionServices` sets a boolean value only.